srtcore/queue.cpp

```diff
@@ -1008,7 +1008,17 @@ void CRendezvousQueue::updateConnStatus(EReadStatus rst, EConnectStatus cst, con
                 // In the below call, only the underlying `processRendezvous` function will be attempting
                 // to interpret these data (for caller-listener this was already done by `processConnectRequest`
                 // before calling this function), and it checks for the data presence.
-                if (!i->m_pUDT->processAsyncConnectRequest(rst, cst, response, i->m_pPeerAddr))
+                bool success;
+                if (i->m_iID != response.m_iID)
+                { 
+                    success = i->m_pUDT->processAsyncConnectRequest(RST_AGAIN, CONN_AGAIN, response, i->m_pPeerAddr);
```

Comment:
I think it only needs to distinguish `cst` parameter. The `rst` parameter is only needed in `processRendezvous` and there you have no possibility that the queue work for two different sockets (connections from other party than the opposite in rendezvous are rejected before this processing can take place). Although I'm not sure how it behaves when you try to make two rendezvous sockets bound to the same port trying to connect to the other endpoint.

In this case you can make the condition directly at the parameter passing place.

Reply:
Ok, in theory you can bind two rendezvous sockets to the same port and even connect them to two different endpoints. Let it better stay as is.